render_silhouette_regmarks.py

```diff
@@ -0,0 +1,131 @@
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA.
+#
+"""
+Base module for rendering regmarks for Silhouette CAMEO products in Inkscape.
+"""
+
+import inkex
+from inkex import EffectExtension, Boolean, Rectangle, PathElement, Layer, Group, TextElement, Transform
+from gettext import gettext
+
+REGMARK_LAYERNAME = 'Regmarks'
+REGMARK_LAYER_ID = 'regmark'
+REGMARK_TOP_LEFT_ID = 'regmark-tl'
+REGMARK_TOP_RIGHT_ID = 'regmark-tr'
+REGMARK_BOTTOM_LEFT_ID = 'regmark-bl'
+REGMARK_SAFE_AREA_ID = 'regmark-safe-area'
+REGMARK_NOTES_ID = 'regmark-notes'
+
+REG_SQUARE_MM = 5
+REG_LINE_MM = 20
+REG_SAFE_AREA_MM = 20
+
+# https://www.reddit.com/r/silhouettecutters/comments/wcdnzy/the_key_to_print_and_cut_success_an_extensive/
+# > The registration mark thickness is actually very important. For some reason, 0.3 mm marks work perfectly. 
+# > The thicker you get, the less accurate registration will be. ~~~ galaxyman47
+REG_MARK_LINE_WIDTH_MM = 0.3
+
+REG_MARK_INFO_FONT_SIZE_PX = 2.5
+
+ENABLE_CHECKERBOARD = True
+
+class InsertRegmark(EffectExtension):
+	def add_arguments(self, pars):
+		# Parse arguments
+		pars.add_argument("-X", "--reg-x", "--regwidth",  type = float, dest = "regwidth",   default = 180.0, help="X mark distance [mm]")
+		pars.add_argument("-Y", "--reg-y", "--reglength", type = float, dest = "reglength",  default = 230.0, help="Y mark distance [mm]")
+		pars.add_argument("--rego-x",  "--regoriginx",    type = float, dest = "regoriginx", default = 15.0,  help="X mark origin from left [mm]")
+		pars.add_argument("--rego-y", "--regoriginy",     type = float, dest = "regoriginy", default = 20.0,  help="X mark origin from top [mm]")
+		pars.add_argument("--verbose", dest = "verbose",  type = Boolean, default = False, help="enable log messages")
+
+	def effect(self):
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else self.svg.to_dimensional(self.svg.get("width"), "mm") - reg_origin_X*2
+		reg_length = self.options.reglength if self.options.reglength else self.svg.to_dimensional(self.svg.get("height"), "mm") - reg_origin_Y*2
+
+		if self.options.verbose == True:
+			self.msg(gettext("[INFO]: page width ")+str(self.svg.to_dimensional(self.svg.get("width"), "mm")))
+			self.msg(gettext("[INFO]: page height ")+str(self.svg.to_dimensional(self.svg.get("height"), "mm")))
+			self.msg(gettext("[INFO]: regmark from document left ")+str(reg_origin_X))
+			self.msg(gettext("[INFO]: regmark from document top ")+str(reg_origin_Y))
+			self.msg(gettext("[INFO]: regmark to regmark spacing X ")+str(reg_width))
+			self.msg(gettext("[INFO]: regmark to regmark spacing Y ")+str(reg_length))
+
+		# Check if existing regmark layer exist and delete it
+		old_regmark_layer = self.svg.getElementById(REGMARK_LAYER_ID)
+		if old_regmark_layer is not None:
```

Comment:
if old_regmark_layer:

Reply:
Attempted to remove, but got this warning message.

```
/usr/share/inkscape/extensions/render_silhouette_regmarks.py:72: FutureWarning: The behavior of this method will change in future versions. Use specific 'len(elem)' or 'elem is not None' test instead.
  if old_regmark_layer:
```

So placing it back in.